### DISCUSSION TITLE
Value and Aux Value for Stat Series Card

### DISCUSSION THREAD

**Discussion by n-srinidhi** [h2oai/wave#1778] [Q&A] [answered]
How to set the values for the tall and wide series stat cards? Have gone through some of the examples in the GitHub repo and documentation, however the format isn't clear. For example, I need to display the count and value using two different wide series stat cards. Also, is it possible to not display an auxiliary values?

**Comment by vopani** [comment, marked as answer]
It'll help if you can provide a min-repro code of the card that doesn't display to what you're looking for.
It's (generally) possible to leave auxiliary value blank.

**Reply by n-srinidhi** [reply]
```
user_stat=ui.wide_series_stat_card(
    box='u_stat',
    title='Users Logged in',
    value='{5}',
    aux_value='={5}',
    plot_data=data(fields=['date','count'], rows=[['22-11-24', 1], ['22-11-25', 2], ['22-11-26', 3]]),
    data={'22-11-26': 3},
    plot_type='area',
    plot_value='count',
    plot_color='$blue',
    plot_zero_value=0,
    plot_curve='linear',
)
```


This is one card I'm trying to display. I need to display an integer value, which is the number of users logged in. The X-Axis will either be a date string or a time value. How do I give the value field?When I try to add `aux_value=''`, so as to skip it, I get a blank page but no errors. If I give `value='5'` so as to display an integer value, I get a blank page again. `value='{5}'` is considered as a string and displayed as follows:


![image](https://user-images.githubusercontent.com/117831190/210509148-30e0429c-2441-4681-998e-e3e207c85089.png)


This is another example I'm trying to understand:
```
value='=${{intl qux minimum_fraction_digits=2 maximum_fraction_digits=2}}',
aux_value='={{intl quux style="percent" minimum_fraction_digits=1 maximum_fraction_digits=1}}',

```

**Reply by vopani** [reply]
Thanks for sharing the example. I struggled with it as well.

IMO, `value='5'` and `aux_value=''` should be supported but it isn't, even I get the blank page. Maybe open an issue and @mturoci can take a look at it unless we're missing something or if there is a good reason it shouldn't be allowed?

@n-srinidhi As a (temporary) workaround, `value=' 5'` and `aux_value=' '` will work for your requirement, but its a hack. I'm not even sure why this works 😬

**Reply by n-srinidhi** [reply]
Haha ok! Thanks a lot! Will try this and open an issue as well

**Reply by mturoci** [reply]
This was a regression on our side, caused by https://github.com/h2oai/wave/issues/1592 (cc @marek-mihok). See [data-binding docs](https://wave.h2o.ai/docs/expressions/#formatting-numbers) for more info on how to use these expressions. In the meantime, you can work around this problem by labeling your data properly:

```py
user_stat=ui.wide_series_stat_card(
    box='u_stat',
    title='Users Logged in',
    value='={{my_value}}',
    aux_value='={{my_value}}',
    plot_data=data(fields=['date','count'], rows=[['22-11-24', 1], ['22-11-25', 2], ['22-11-26', 3]]),
    data={'my_value': 5}, # This is used for feeding the value/aux_value expressions.
    plot_type='area',
    plot_value='count',
    plot_color='$blue',
    plot_zero_value=0,
    plot_curve='linear',
)
```

Nonetheless, the primitive, non-expression values should be supported as well. Fixed in https://github.com/h2oai/wave/commit/0c13f33ba7828ed4e9642de176c1736eb8a8378c

**Reply by n-srinidhi** [reply]
Thanks!
